src/modules/resource/inventory.c

```diff
@@ -489,7 +507,17 @@ static void resource_get_xml (flux_t *h,
         errstr = "this RPC only works on rank 0";
         goto error;
     }
+    /* xml is still being collected.  If broker.quorum is not the entire
+     * instance, then this may never complete, so fail the request in that
+     * case; otherwise save the request in inv->waiters for later response.
+     */
     if (!inv->xml) {
+        if (quorum_subset (h)) {
+            errno = EINVAL;
```

Comment:
`EWOULDBLOCK` or `EAGAIN`? Since it is not like the RPC argument was invalid but the state of the flux instance cannot service the RPC?

Reply:
Given that we're returning a textual error, all we really need for errno is _something_ nonzero.  The intent was that fluxion would detect the error and log the human readable string at LOG_ERR level to leave a big clue for someone.  It shouldn't need to do anything with the errno value.

Reply:
Fluxion use case is fine. Just caught my eyes in case this RPC used by something other than fluxion in the future. Ok as is.